typespec-extension/src/code-model-builder.ts

```diff
@@ -2062,8 +2062,16 @@ export class CodeModelBuilder {
       }
     } else if (isRecordModelType(this.program, type)) {
       // "pure" Record processed elsewhere
+
       // "mixed" Record that have properties, treat the model as "additionalProperties"
-      const parentSchema = this.processDictionarySchema(type, this.getName(type));
```

Comment:
Bug was that previously this `type` get cached as a `DictionarySchema` in this line.

It should be cached as an `ObjectSchema`, while its sourceModel be potentially `DictionarySchema`